Tests/GRPCHTTP2CoreTests/Server/GRPCServerStreamHandlerTests.swift

```diff
@@ -26,13 +26,14 @@ import XCTest
 @available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
 final class GRPCServerStreamHandlerTests: XCTestCase {
   func testH2FramesAreIgnored() throws {
+    let channel = EmbeddedChannel()
```

Comment:
Can we have a test which validates that the promise is succeeded when we expect? And another which checks that its failed if the handler is immediately removed from the pipeline?